net/nimble/host/src/ble_hs_hci_cmd.c

```diff
@@ -173,13 +173,20 @@ ble_hs_hci_cmd_body_le_set_adv_params(const struct hci_adv_params *adv,
         return -1;
     }
 
+/* When build with nimBLE controller we know it is BT5 compliant so no need
+ * to limit non-connectable advertising interval
+ */
+#if MYNEWT_VAL(BLE_DEVICE)
```

Comment:
Maybe we should test also for BSP since it will not apply for nrf51?
For sure  it will work for MYNEWT_VAL(BSP_NRF52840)

Reply:
hmm for build in nimble it is ok because controller does same checks. So maybe we could remove more checks from host then?